angular/src/app/landing/landing.component.html

```diff
@@ -144,20 +145,23 @@ <h2>{{record.title}}</h2>
             <div class="ui-g-4 ui-md-4 ui-lg-4 ui-sm-12">
               <span *ngIf="HomePageLink">
                 <a [(href)]="record.landingPage" target="_blank" style="float: right;">
-                  <button pButton type="button" class="homePageLink" icon="faa faa-external-link"
-                    label="Visit Home Page"></button>
+                  <span *ngIf="inBrowser; else visitHomeOnServer">
+                    <button pButton type="button" class="homePageLink" icon="faa faa-external-link" label="Visit Home Page"></button>
+                  </span>
+                  <ng-template #visitHomeOnServer>Visit Home Page</ng-template>
                 </a>
               </span>
             </div>
           </div>
           <div class="ui-g">
             <div class="ui-g-12 ui-md-12 ui-lg-12 ui-sm-12">
               <description-resources [record]="record" [files]="files" [distdownload]="distdownload"
-                [isInternal]="isInternal" [filescount]="filescount" [metadata]="metadata"></description-resources>
+                                     [isInternal]="isInternal" [filescount]="filescount" [inBrowser]="inBrowser"
+                                     [metadata]="metadata"></description-resources>
               <metadata-detail [record]="record" [serviceApi]="serviceApi" [metadata]="metadata"></metadata-detail>
```

Comment:
Should we add [inBrowser]="inBrowser" here as well?

Reply:
I understand that. I meant line 157 - I saw the MetadataComponent had @Input() inBrowser but seems the param was missing in line 157...

Reply:
Ah, I see--sorry about that!  (The display did not make it obvious which line you were talking about.)  

Yes, you are right!!  It should be there; I'll add it real quick.  Thanks!

Reply:
I enjoy reading your code :) very helpful for understanding server side rendering.

Reply:
Thanks for the careful review!